Feature dashboard skeleton

amha19 opened the pull request:
Draft: 
Added navbar component with profile dropdown
Added link to the menu list
Setup global context
Sidebar and content page not complete

amha19 committed:
Added navbar to dashboard

amha19 committed:
Dropdown for profile added.

Added h6 theme and moved the navbar styles to its own file.

amha19 committed:
Added link to menu list

amha19 committed:
Setup global context

Setup a global context in combination  with useReducer.

jessdesr commented on client/src/pages/dashboard/Dashboard.js:
```diff
@@ -0,0 +1,55 @@
+
+const useStyles = makeStyles(theme => ({
+    sidebar: {
+        height: '85vh',
```
Is this to keep the sidebar from going under the navbar?  If so, you should be able to structure the layout to avoid the need for this.

amha19 replied:
No, this is a temporary height to see 'flexGrow' properties of the inside divs, since I haven't put the actual content in there yet the grid shrinks to the height of the sample texts.

jessdesr commented on client/src/pages/dashboard/Dashboard.js:
```diff
@@ -0,0 +1,55 @@
+        <Grid>
+            <Navbar />
+            <Grid container>
+                <Grid
```
You can basically wrap up this skeleton by moving Sidebar out into its own component!

jessdesr commented on client/src/pages/layout/Navbar.js:
```diff
@@ -0,0 +1,135 @@
+    return (
+        <div className={classes.root}>
+            <AppBar position="static" color="default" elevation={2}>
+                <Toolbar style={{ height: '100px', backgroundColor: '#FFF' }}>
```
Try to avoid inline styling wherever you can: I know it's appealing when building the layout but can get easily lost in the code when also combined with useStyles.

amha19 replied:
I have removed the inline styles.

amha19 committed:
Fixed in-line styling

amha19 committed:
Created empty Sidebar, content pages component

Created protected routes so that if user is not authenticated it will redirect to login

amha19 committed:
Merge branch 'dev' into feature-dashboard-skeleton-#4

amha19 committed:
Merge branch 'dev' into feature-dashboard-skeleton-#4
